dev/RadioButtons/RadioButtons.cpp

```diff
@@ -115,8 +112,17 @@ void RadioButtons::OnRepeaterLoaded(const winrt::IInspectable&, const winrt::Rou
             AttachToLayoutChanged();
         }
 
-        UpdateSelectedItem();
-        UpdateSelectedIndex();
+        m_blockSelecting = false;
+        if (ReadLocalValue(s_SelectedIndexProperty) == winrt::DependencyProperty::UnsetValue()
+            && ReadLocalValue(s_SelectedItemProperty) != winrt::DependencyProperty::UnsetValue())
+        {
+            UpdateSelectedItem();
+        }
+        else
```

Comment:
What does SelectedIndex() return if ReadLocalValue(s_SelectedIndexProperty)==winrt::DependencyProperty::UnsetValue()? The default -1?

Reply:
Do we have to use ReadLocalValue, or would just checking the properties versus defaults work? I ask because ReadLocalValue is a superpower that I wish we didn't have. It relies on behavior of DependencyProperty storage that I hope we can move away from in the future.

Reply:
Agreed, we shouldn't have any behavior that depends on whether or not a local value is set.

Reply:
Yes, but ReadLocalValue(s_SelectedIndexProperty)==winrt::DependencyProperty::UnsetValue() will be false if the consumer has set the value to the default value of -1 as well.

Reply:
Interesting, no I don't think we need to use it, but distinguishing between the default value and a user provided value which is equivalent to the default seems useful to me.

Reply:
Can you rewrite so it doesn't use ReadLocalValue? For example if someone set it and then it got set back to the default it wouldn't be UnsetValue again. I'd feel more confident about the logic if it worked just against the plain property getters.

Reply:
I think i already did